pymatgen/io/lammps/data.py

```diff
@@ -865,6 +863,36 @@ def from_structure(cls, structure: Structure, ff_elements=None, atom_style="char
         topo = Topology(boxed_s)
         return cls.from_ff_and_topologies(box=box, ff=ff, topologies=[topo], atom_style=atom_style)
 
+    def set_charge_atom(self, charges: dict):
+        """
+        Add or modify charges of atoms of the data.
+
+        Args:
+            charges: Dict containing the charges for the atoms to set.
+                     The dict should contain atom indexes and charges.
+                     Example: change the charge of the third atom of the list to -2:
+                     charges={3: -2}
+        """
```

Comment:
This doesn't actually add a command to the LAMMPS input, but it rather changes the data file directly. The "set charge" command is used in the input when the data does not contain information (or not the right information) on the charge. 
 But indeed, the 3rd index is not necessarily the 3rd atom in the list, it was the case when I was including some other changes that I removed since. 
The type hints from typing are not that much used in pymatgen, so I don't know if we should start using them or not?